### PR TITLE
login: switch to http GET for msi login and bring back the global cred cache 

### LINKED ISSUE #5465: test: stub out the client factory creation before run login unit tests
Found out when code reviewed the fix for #5464 
Looks like the test seam is no longer working so running unit tests could wipe out the real azureProfile.json under `~/.azure`. Very bad.

### PR DESCRIPTION
**Pull request by yugangw-msft**
The context is the IMDS we need to migrate in March only supports GET, also the cloud shell which will leverage CLI's msi login supports GET as well for token retrieval, so to make the code migration easier, I am just switching to GET so the same flow can be used for both MSI and cloud shell authentications.
This also fix #5465; and also rewires the global creds cache (needed by datalake service for better perf 
 for large file transfer) which appeared lost during knack conversion

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [na] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5708
This is an **experimental preview** for @microsoft users.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -105,13 +105,25 @@ class CredentialType(Enum):  # pylint: disable=too-few-public-methods
...
-        self._creds_cache = CredsCache(self.cli_ctx, self.auth_ctx_factory, async_persist=async_persist)
+
+        if use_global_creds_cache:
+            # for perf, use global cache
+            if not Profile._global_creds_cache:
+                Profile._global_creds_cache = CredsCache(self.cli_ctx, self.auth_ctx_factory,
+                                                         async_persist=async_persist)
+            self._creds_cache = Profile._global_creds_cache
+        else:
+            self._creds_cache = CredsCache(cli_ctx, self.auth_ctx_factory, async_persist=async_persist)
```
Comment (williexu) [inline src/azure-cli-core/azure/cli/core/_profile.py]: `CredsCache(self.cli_ctx, ....)`

File: src/azure-cli-core/azure/cli/core/tests/test_profile.py [hunk 2]
```diff
@@ -741,89 +741,6 @@ def test_get_login_credentials_for_graph_client(self, mock_get_token, mock_read_
...
-            "environmentName": "AzureCloud",
-            "tenantId": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a"
-        }
-
-        actual = get_file_json(test_token_file)
-
-        self.assertEqual([expected_subscription], result_accounts)
-        # sanity check that the expiration time is about 45 minutes away
-        self.assertTrue(parser.parse(actual[0]['expiresOn']) < datetime.now() + timedelta(minutes=45))
-        self.assertTrue('oid' not in actual[0])
```
Comment (williexu) [inline src/azure-cli-core/azure/cli/core/tests/test_profile.py]: why is this test being removed?

File: src/azure-cli-core/azure/cli/core/_profile.py (hunk 1 above)
Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/_profile.py]: Okay. I assume you like for consistency, as the current code still functions

File: src/azure-cli-core/azure/cli/core/tests/test_profile.py (hunk 2 above)
Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/tests/test_profile.py]: The cloud shell login flow this test targets is gone as we are moving to msi model, hence i just remove the test

File: src/azure-cli-core/azure/cli/core/_profile.py (hunk 1 above)
Comment (williexu) [reply src/azure-cli-core/azure/cli/core/_profile.py]: this wouldn't pick up the result of `get_default_cli()` if no cli_ctx was passed in

Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/_profile.py]: Good find. I didn't realize that line was touched by knack as well. This branch so far only gets used by tests, so the error has not surfaced.

**Pull request merged**